django2_codemods/widget_add_renderer_mod.py

```diff
@@ -0,0 +1,41 @@
+import sys
+from lib2to3.fixer_util import Name, KeywordArg, Comma
+from typing import Optional
+
+from bowler import Query, LN, Capture, Filename, TOKEN
+from fissix.pytree import Leaf
+
+
+def render_has_no_renderer(node: LN, capture: Capture, filename: Filename) -> bool:
+
+    if 'function_call' in capture:
+        return False  # This is a function call, no need to add argument
+
+    arguments = capture.get("function_arguments")[0].children
+    for arg in arguments:
+        if arg.type == TOKEN.NAME and arg.value == "renderer":
+            return False  # This definition already has a renderer argument.
+
+    return True
+
+
+def add_renderer(node: LN, capture: Capture, filename: Filename) -> Optional[LN]:
+    arguments = capture.get("function_arguments")[0]
+    new_renderer_node = KeywordArg(Name(" renderer"), Name("None"))
+
+    if isinstance(arguments, Leaf):  # Node is a leaf and so we need to replace it with a list of things we want instead
+        arguments.replace([arguments.clone(), Comma(), new_renderer_node])
+    else:
+        arguments.append_child(Comma())
+        arguments.append_child(new_renderer_node)
+
+    return node
+
+
+(
+    Query(sys.argv[1])
+        .select_method("render")
```

Comment:
`render()` is a somewhat common method name; would it be much extra work to additionally filter to just `render` methods declared in in descendants of Django's `Widget` class?  I think this would be a usage of [select()](https://pybowler.io/docs/api-selectors#select) with a custom pattern that blends the ones used in in `select_subclass()` and `select_method()`, found in https://github.com/facebookincubator/Bowler/blob/master/bowler/query.py .

Reply:
Got it.